### PR TITLE
Fixes #135: Handle application configuration using `QSettings`

### LINKED ISSUE #135: Handle application configuration using `QSettings`
**Description**
We are using JSON format to save/restore configuration, while using `QSettings` is a better approach.

**Advantages over the JSON approach**
- Uses the standard configuration format for every system the application runs on.
  - Windows: System registry
  - Darwin: Property list files
  - UNIX: In absence of an standard, uses INI text files
- Creating and destroying `QSettings` objects is very fast
- It ensures that the location is **always** writable
- The location and storage format is transparent for the developer
- Each developer can save/load its own settings with two lines of code without affecting others or making a bigger implementation
- There is an equivalent QML component named `Settings` that allows to write and read settings directly from QML

*See `QSettings` and `Settings` in Qt Docs for details*

**How to use it?**

Read configurations when necessary:

```c++
QSettings settings;

int value1 = settings.value("cipher/someInteger").toInt();
QString value2 = settings.value("cipher/someString").toString();
```

And write/update configurations when necessary:

```c++
QSettings settings;

int value1 = 5;
QString value2 = "bla bla bla";
settings.setValue("cipher/someInteger", value1);
settings.setValue("cipher/someString", value2);
```

In the above examples, "cipher" it's just a group of properties, like a folder.
You can name your group (if any) as you want.

### LINKED ISSUE #284: Custom QT Docker image for testing
**Is your feature request related to a problem? Please describe.**
The Docker image `therecipe/qt:linux` seems to [include an old version of `git`](https://travis-ci.org/fibercrypto/fibercryptowallet/jobs/625358890#L367-L421).

**Describe the solution you'd like**
Derive a new image with git upgraded and use it for running test coverage on Travis CI

**Describe alternatives you've considered**
This can be done in `make run-docker` but has two main drawbacks. That would slow down the builds and over-complicate the command needed to make this work. 

**Additional context**
Discovered while working on #260 for #135 .

### PR DESCRIPTION
**Pull request by mauricio1802**
Fixes #135 
Fixes #284 

**Changes:**
- Change the approach used for the app settings
- Create a Config system in the backend allowing to plugins to define their own settings in a plugable way
- QSettings is used for accomplish the persisten over multiple platform
- Add `make test-docker` target to run commands inside `thereciepe/qt:linux` container
- Run `make test-docker` on Travis CI since `QSettings` tests rely on `therecipe/qt`

**Does this change need to mentioned in CHANGELOG.md?**
 no

**Requires testing**
yes

### CODE REVIEW

**Comment by olemis** [comment]
@stdevMauricio1802 is this ready to be merged ?

**Review by olemis — CHANGES_REQUESTED** [review]
Needs test cases

File: src/models/config.go [hunk 1]
```diff
@@ -1,85 +1,179 @@
...
 	qtcore.QObject
-	_ int    `property:"sourceType"`
-	_ string `property:"source"`
+	configManager *local.ConfigManager
+	_             func()                      `constructor:"init"`
+	_             func(string, string)        `slot:"setValue"`
+	_             func(string) string         `slot:"getValue"`
+	_             func() []string             `slot:"getSections"`
+	_             func(string) *ConfigSection `slot:"getSection"`
 }
```
Comment (stdevCrow) [inline src/models/config.go]: We need a way to get the default values. Something like a `getDefault("skycoin/walletSource/1/Source")` should return `"/home/username/.skycoin/wallets"`.

**Review by stdevCrow — CHANGES_REQUESTED** [review]

**Comment by stdevCrow** [comment]
@stdevMauricio1802 `Apply` and `Discard` settings work as expected, but `Restore defaults` don't. Please check the requested changes. Once done we can merge and close this PR.

File: src/models/config.go (hunk 1 above)
Comment (olemis) [reply src/models/config.go]: @stdevMauricio1802 already added that feature . can this conversation be marked as resolved ?

**Review by stdevCrow — APPROVED** [review]

**Pull request merged**